### ISSUE TITLE
Lambda parameters for @Nullable are incorrect

### ISSUE THREAD

**Issue by eirnym** [typetools/checker-framework#3217]
The following code produces false positive with types changed quite dramatically in terms of nullness. 

#### Code:
```
import org.checkerframework.checker.nullness.qual.Nullable;

import java.util.function.Function;

public class Main<ModelA, ModelB, Value> {
    private final Function<Function<ModelA, @Nullable Value>, Function<ModelB, @Nullable Value>> proxyFunction;

    public Main(Function<Function<ModelA, @Nullable Value>, Function<ModelB, @Nullable Value>> proxyFunction) {
        this.proxyFunction = proxyFunction;
    }
}

public class SubClass<ModelA, Value> extends Main<ModelA, ModelA, Value> {
    public SubClass() {
        super(x -> x);
    }
}
```

#### The output of checker framework is:
```
> Task :example1:compileJava
/home/eirnym/t/example1/src/main/java/example1/SubClass.java:5: error: [lambda.param.type.incompatible] incompatible parameter types in lambda expression.
        super(x -> x);
              ^
  found   : @Initialized @NonNull Function<ModelA[ extends @Initialized @Nullable Object super @Initialized @NonNull Void], Value[ extends @Initialized @Nullable Object super @Initialized @Nullable Void]>
  required: @Initialized @NonNull Function<ModelA[ extends @Initialized @Nullable Object super @Initialized @NonNull Void], Value[ extends @Initialized @Nullable Object super @Initialized @NonNull Void]>
/home/eirnym/t/example1/src/main/java/example1/SubClass.java:5: error: [return.type.incompatible] incompatible types in return.
        super(x -> x);
                   ^
  type of expression: @Initialized @NonNull Function<ModelA[ extends @Initialized @Nullable Object super @Initialized @NonNull Void], Value[ extends @Initialized @Nullable Object super @Initialized @NonNull Void]>
  method return type: @Initialized @NonNull Function<ModelA[ extends @Initialized @Nullable Object super @Initialized @NonNull Void], Value[ extends @Initialized @Nullable Object super @Initialized @Nullable Void]>
2 errors
```
#### My Environment is:
* Gradle 6.3
* Java 8 & Java 11
* Gradle CheckerFramewrok plugin 0.4.13
* CheckerFramework 3.2.0
* Arch Linux

#### Additional notes:
You can also see the code and sample project in a [repository](https://github.com/eirnym/checkerframework-falsepositives/tree/master/example1) in module `example1`.
